csrc/mmdeploy/backend_ops/tensorrt/roi_align/trt_roi_align.cpp

```diff
@@ -203,9 +203,9 @@ nvinfer1::IPluginV2 *TRTRoIAlignCreator::createPlugin(
       int data_size = fc->fields[i].length;
       const char *data_start = static_cast<const char *>(fc->fields[i].data);
       std::string poolModeStr(data_start, data_size);
-      if (poolModeStr == "avg") {
+      if (strcmp(poolModeStr.c_str(), "avg") == 0) {
```

Comment:
operator '==' is done via `string::compare()`.
For checking whether two strings are the same (true or false) instead of comparing which is greater or not, I think `==` is fine.

Reply:
poolModeStr is a std::string with data "avg\0"

Reply:
https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp
> Compares the contents of a string with another string or a null-terminated array of CharT.

Reply:
`the contents of a string` contains a null byte. The comparison will always be failed (Yes, it did in the unit test).
If you really prefer the operator, I will cast the c-string to a std::string.

Reply:
Could you show me your UT?
here is what I did
```C++
#include <iostream>

#include <string>


int main() {
  std::string pool_mode("avg");
  std::cout << (pool_mode == "AVG") << ", " << (pool_mode == "avg") << std::endl;
  return 0;
}

```
The output is: 0, 1

Reply:
This is THE string.
```c++
char data[4] = {'a','b','c','\0'};
string pool_mode(pool_mode, 4);
```

Reply:
Why not using `string pool_mode(data)`?

Reply:
> This is THE string.
> 
> ```c++
> char data[4] = {'a','b','c','\0'};
> string pool_mode(pool_mode, 4);
> ```

This initialization indeed treats `pool_mode` as a binary buffer rather than a string

Reply:
Yes, it did.
https://github.com/open-mmlab/mmdeploy/blob/e76c0a8a8ba24b1254ef299d2efc6bf9feeb57b8/csrc/mmdeploy/backend_ops/tensorrt/roi_align/trt_roi_align.cpp#L203-L208

The length varies between different version, what can I do?

Reply:
> Why not use `string pool_mode(data)`?

https://github.com/open-mmlab/mmdeploy/blob/e76c0a8a8ba24b1254ef299d2efc6bf9feeb57b8/csrc/mmdeploy/backend_ops/tensorrt/roi_align/trt_roi_align.cpp#L203

The data is read from the serialized data. And the length varies between different versions. What if they decide to omit null bytes as long as they already provide the data length?